core/sql/generator/GenExplain.cpp

```diff
@@ -660,6 +660,269 @@ FileScan::addSpecificExplainInfo(ExplainTupleMaster *explainTuple,
   return(explainTuple);
 }
 
+static void appendListOfColumns(Queue* listOfColNames,ComTdb *tdb, NAString& outNAString){
+
+	if (((ComTdbHbaseAccess*)tdb)->sqHbaseTable()){// if trafodion table
+	  char buf[1000];
+
+	  listOfColNames->position();
+	  for (Lng32 j = 0; j < listOfColNames->numEntries(); j++)
+	    {
+	      char * currPtr = (char*)listOfColNames->getCurr();
+
+	      Lng32 currPos = 0;
+	      Lng32 jj = 0;
+	      short colNameLen = *(short*)currPtr;
+	      currPos += sizeof(short);
+	      char colFam[100];
+	      while (currPtr[currPos] != ':')
+		{
+		  colFam[jj] = currPtr[currPos];
+		  currPos++;
+		  jj++;
+		}
+	      colFam[jj] = ':';
+	      jj++;
+	      currPos++;
+	      colFam[jj] = 0;
+	      colNameLen -= jj;
+
+	      NABoolean withAt = FALSE;
+	      char * colName = &currPtr[currPos];
+	      if (colName[0] == '@')
+		{
+		  colNameLen--;
+		  colName++;
+		  withAt = TRUE;
+		}
+
+	      Int64 v;
+	      if (colNameLen == sizeof(char))
+		v = *(char*)colName;
+	      else if (colNameLen == sizeof(unsigned short))
+		v = *(UInt16*)colName;
+	      else if (colNameLen == sizeof(Lng32))
+		v = *(ULng32*)colName;
+	      else
+		v = 0;
+	      if (j==0)
+		      str_sprintf(buf, "%s%s%Ld",
+				  colFam,
+				  (withAt ? "@" : ""),
+				  v);
+	      else
+			  str_sprintf(buf, ",%s%s%Ld",
+				  colFam,
+				  (withAt ? "@" : ""),
+				  v);
+
+	      outNAString += buf;
+
+	      listOfColNames->advance();
+	    } // for
+	}// trafodion tables
+	else
+	{// if hbase native tables
+	  char buf[1000];
+
+	  listOfColNames->position();
+	  for (Lng32 j = 0; j < listOfColNames->numEntries(); j++)
+	    {
+	      char * currPtr = (char*)listOfColNames->getCurr();
+
+	      char * colNamePtr = NULL;
+
+		  Lng32 currPos = 0;
+		  short colNameLen = *(short*)currPtr;
+		  currPos += sizeof(short);
+		  char colName[500];
```

Comment:
yes. use of snprintf() will prevent overrun. We once did this as an exercise to change all instances of non-snprintf() to snprintf(). It is a requirement now to use snprintf().